### DISCUSSION TITLE
RTL version of the current bootScore version

### DISCUSSION THREAD

**Discussion by matanmn33** [bootscore/bootscore#301] [Q&A]
Hey,
I would like to know if this version supports RTL and how? should i replace the main.css with bootstrap rtl version?
Built-in option?
Thanks!

**Comment by crftwrk** [comment]
Hi,

this is a really good question. If you just replace Bootstrap with precompiled Bootstrap rtl in `main.css`, it will be overridden with ltr by scssphp each time when you change something in your scss files. Unfortunately scssphp doesn't support rtl https://github.com/bootscore/bootscore/issues/181#issuecomment-1173652769.

However, rtl is something which is still on the list. Maybe @ggggg has a solution for it?

It is not possible to override ltr in child in a simple way. The only way I see here for now is changing the directions manually in main-theme's source. For example changing left-right in https://github.com/bootscore/bootscore/blob/main/scss/bootstrap/_utilities.scss

Default ltr
```scss
"margin-start": (
  responsive: true,
  property: margin-left,
  class: ms,
  values: map-merge($spacers, (auto: auto))
),
```

into rtl

```scss
"margin-start": (
  responsive: true,
  property: margin-right,
  class: ms,
  values: map-merge($spacers, (auto: auto))
),
```

Editing the main-theme is not a good idea, but if we know which files have to be changed we can provide examples and add them to the docs. At first glance this affects 3 files:

- https://github.com/bootscore/bootscore/blob/main/scss/bootstrap/_utilities.scss
- https://github.com/bootscore/bootscore/blob/main/scss/bootstrap/forms/_input-group.scss
- https://github.com/bootscore/bootscore/blob/main/scss/bootstrap/_button-group.scss

This job should be done by a real rtl user. If you want to help here, we look forward to your contribution.

**Reply by matanmn33** [reply]
From yesterday to now i tried to figure out how to handle that thing and for the RTL version i found a pretty simple and nice solution but i'm not sure this solution will continue forever. 
Thus, average user and developer would not change the scss files every time and when he/she done with the scss files and customization it will stay as is for a period, so the solution i found looks like this: Finish changing and working with the scss files and variables and when the main.css is compiled i just take it to RTLCSS platform and render it as RTL main.css file so it will stay like this and unfortunately every time i change the scss files and the main.css compiled i need to redo the think with RTLCSS.
for the time being i could not find another solution.

**Reply by crftwrk** [reply]
Another possibility is to disable scssphp in main-theme and use Gulp or any other built-tool. There many bootScore rtl sites out there, but no rtl user has explained how he solved this so far.

**Reply by matanmn33** [reply]
I found scssphp more useful here.
Change whatever you want with the scss files -> the compiled main.css copy-paste here https://rtlcss.com/playground/ and after that just replace the output from rtlcss right back to main.css and done!
Btw, i can help and translate bootScore to Hebrew (RTL of course).
Thanks for commenting and help me to figure things out.

**Reply by crftwrk** [reply]
Thank you for sharing your approach. We will provide a integrated rtl solution in future when we know how to solve it.

> Btw, i can help and translate bootScore to Hebrew (RTL of course).

Hebrew is already there https://github.com/bootscore/bootscore/blob/main/languages/he_IL.po. If you have ideas for improvements, you're always welcome.

**Reply by ggggg** [reply]
Hi, PHP and css really aren't my strong suits but I'll  have a closer look at the solutions I implemented to make my project work a few months ago.
@matanmn33 my hebrew isn't great so if you could go over my translations that would be much appreciated. Plus a few admin translations are missing so if you can, please fill those out as well.
